### PR TITLE
Bugfix/fix modification observer disable

### PR DESCRIPTION
**Pull request by hata6502**
https://github.com/codex-team/editor.js/issues/1339

### CODE REVIEW

File: src/components/modules/modificationsObserver.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export default class ModificationsObserver extends Module {
   /**
    * Allows to temporary disable mutations handling
    */
-  private disabled: boolean;
+  private disabled = false;
```
Comment (hata6502) [inline src/components/modules/modificationsObserver.ts:31]: I use strict mode of TypeScript. 
https://github.com/impact-blue/editor.js/pull/21#discussion_r495694650

Changes of this file are not related to https://github.com/codex-team/editor.js/issues/1339, 
but these changes are more safety for TypeScript.

File: src/components/modules/saver.ts [hunk 2]
```diff
@@ -36,16 +36,18 @@ export default class Saver extends Module {
...
+      });
 
-    ModificationsObserver.enable();
+      const extractedData = await Promise.all(chainData);
+      const sanitizedData = await Sanitizer.sanitizeBlocks(extractedData);
 
-    return this.makeOutput(sanitizedData);
+      return this.makeOutput(sanitizedData);
+    } finally {
+      ModificationsObserver.enable();
```
Comment (hata6502) [inline src/components/modules/saver.ts:49]: Even throws an exception, enable modification observer finally.

File: src/components/modules/modificationsObserver.ts [hunk 3]
```diff
@@ -37,7 +37,10 @@ export default class ModificationsObserver extends Module {
    */
   private mutationDebouncer = _.debounce(() => {
     this.updateNativeInputs();
-    this.config.onChange(this.Editor.API.methods);
+
+    if (this.config.onChange) {
```
Comment (gohabereg) [inline src/components/modules/modificationsObserver.ts:41]:
```suggestion
    if (typeof this.config.onChange === 'function') {
```

**Comment by hata6502** [comment]
This is a critical fix when Editor.js saves the block data in onChange callback. 
Would you merge it in v2.19.0? 🙏

```
onChange: async () => {
  // Block data saving is unstable because save method may throws an exception. 
  const data = await editorJS.save();

  localStorage.setItem('block-data', data);
}
```

**Review by gohabereg — APPROVED** [review]

**Review by robonetphy — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Comment by neSpecc** [comment]
@hata6502 please, resolve conflicts

**Comment by neSpecc** [comment]
Can you open pushes for us?

**Comment by hata6502** [comment]
Conflict resolved. 

I use GitHub Organization, so I can't allow edits from maintainers. 
https://qiita.com/ryysud/items/4d31dd86efab34a7fbaf#github-%E3%82%B5%E3%83%9D%E3%83%BC%E3%83%88%E3%81%8B%E3%82%89%E3%81%AE%E5%9B%9E%E7%AD%94

File: src/components/modules/saver.ts [hunk 4]
```diff
@@ -36,16 +36,18 @@ export default class Saver extends Module {
...
+        chainData.push(this.getSavedData(block));
+      });
 
-    ModificationsObserver.enable();
+      const extractedData = await Promise.all(chainData);
+      const sanitizedData = await Sanitizer.sanitizeBlocks(extractedData);
 
-    return this.makeOutput(sanitizedData);
+      return this.makeOutput(sanitizedData);
+    } finally {
```
Comment (ranemihir) [inline src/components/modules/saver.ts:48]: why isn't there a catch statement for this try block?

Comment (hata6502) [reply src/components/modules/saver.ts:48]: Because

```
try {
  // ...
} catch (e) {
  throw e;
} finally {
  // ...
}
```

equals

```
try {
  // ...
} finally {
  // ...
}
```
